src/tallies/tally_filter_zernike.F90

```diff
@@ -148,6 +232,10 @@ function openmc_zernike_filter_get_params(index, x, y, r) result(err) bind(C)
         x = f % x
         y = f % y
         r = f % r
+      type is (ZernikeRadialFilter)
```

Comment:
Use `class is (ZernikeFilter)` instead of `type is` above and you can remove this block entirely (since `ZernikeRadialFilter` is of class `ZernikeFilter`